### ISSUE TITLE
Question regarding Comparator.comparing()

### ISSUE THREAD

**Issue by ghost** [nus-cs2103-AY1819S1/forum#7]
So I have a question regarding `Comparator.comparing()`. More specifically, how exactly to use it?

This is because in my `sortAddressBook()` method in ABL1, which is classified under

>  INTERNAL ADDRESS BOOK DATA METHODS

, I wrote the code body inside the method in such a way where IntelliJ told me that Can be replaced with `Comparator.comparing`.

By the way, I also used an array list of hash maps, where each entry in the array list represents the data of a single person (name, phone and email). And I used enumerations to store the aforementioned data (name, phone and email).

I know that perhaps this is more of a "technical" or coding question which maybe none of the TAs or teaching staff are allowed to "teach" me. Nonetheless, if there are any kind souls who can help without "spoon-feeding", I will sincerely appreciate from the bottom of my heart :) .

In the meantime, I will try to figure it out myself in the near future when I do my research and scouring again :D

**Comment by azhikai** [comment]
Refer to this https://www.baeldung.com/java-comparator-comparable. You may wish to note the difference between `Comparator` and `Comparable` while reading the website as people (like me) tend to mix them both up.

Based on what you said, Java is recommending you to refactor whatever you wrote to `Comparator.comparing()` instead which is a Java 8 feature. You may refer to https://www.baeldung.com/java-8-comparator-comparing for more information. 

In short, your comparator might end up becoming something like `Comparator.comparing(x -> x...)`, where `x` is the param used by the lambda expression `x -> x...` that determines the sort field -- in this case, you can sort by any of your enum value like this:

```java
    private static Type sort() {
        list.sort(Comparator.comparing(x-> x.get(ENUM_TYPE)));
        return;
    }
```

Hope this helps.

**Comment by ghost** [comment]
Thank you very much @AZhiKai for the rather informative post! I will check it out at another time as it is late :).

> You may wish to note the difference between Comparator and Comparable

Yes, another is the `Collection` and` Collections` classes in Java API.

Also, yes, I did read up a bit on the `Comparator` class, and more specifically about the `Comparator.comparing()` method. However, my priority was on finishing the LO which earns participation marks~

Another thing I think that I wish to learn about and familiarise myself with is about Lambda expressions.
